File: Notebooks/resources/python_vs_R.R
library(readr)
library(nnet)
library(MASS)
library(caret)


main <- function() {
  
  header <- c("sepal_length", "sepal_width", "petal_length", "petal_width", "class")
  iris <- read.csv("https://archive.ics.uci.edu/ml/machine-learning-databases/iris/iris.data")
  names(iris) <- header
  sample <- sample.int(n = nrow(iris), size = floor(.8 * nrow(iris)), replace=F)
  train <- iris[sample, ]
  test  <- iris[-sample, ]
  
  # logistic regression
  lr <- multinom(class ~ ., data = train, trace = F)
  
  # linear discriminant analysis
  lda_model <- lda(class ~ ., data = train)
  
  # KNN
  man_grid <- expand.grid(k = c(1:10))
  ctrl <- trainControl(method="cv", number = 5)
  knn <- train(class ~ ., data = train, method = "knn", trControl = ctrl, tuneGrid = man_grid)
  
  # SVM
  man_grid <- expand.grid(cost = c(1:10))
  ctrl <- trainControl(method="cv", number = 5)
  svm <- train(class ~ ., data = train, method = "svmLinear2", trControl = ctrl, tuneGrid = man_grid)
  
  # Evaluate
  lr_accuracy <- sum(predict(lr, test) == test$class) / nrow(test)
  lda_accuracy <- sum(predict(lda_model, test)$class == test$class) / nrow(test)
  knn_accuracy <- sum(predict(knn, test) == test$class) / nrow(test)
  svm_accuracy <- sum(predict(svm, test) == test$class) / nrow(test)
  
  # print(paste(lr_accuracy, ",", lda_accuracy, ",", knn_accuracy, ",", svm_accuracy))
}


now <- Sys.time()
for (i in c(1:5)) {
  main()
}
print(Sys.time() - now)


a = 2
b <- 2
b

File: Notebooks/resources/python_vs_R.py
import pandas as pd
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LogisticRegression
from sklearn.discriminant_analysis import LinearDiscriminantAnalysis
from sklearn.model_selection import GridSearchCV, KFold
from sklearn.neighbors import KNeighborsClassifier
from sklearn import svm

def main():

    names = ["sepal_length", "sepal_width", "petal_length", "petal_width", "Name"]
    iris_data = pd.read_csv("https://archive.ics.uci.edu/ml/machine-learning-databases/iris/iris.data", names = names)
    train, test = train_test_split(iris_data, test_size=0.2)

    X_train = train.drop('Name', axis=1)
    y_train = train['Name']
    X_test = test.drop('Name', axis=1)
    y_test = test['Name']

    # logistic regression
    lr = LogisticRegression(solver='lbfgs', multi_class='auto', max_iter=1000)
    lr.fit(X_train, y_train)

    # linear discriminant analysis
    lda = LinearDiscriminantAnalysis()
    lda.fit(X_train,y_train)

    # KNN (k-nearest neighbours)
    parameters = {'n_neighbors': range(1,11)}
    knn = GridSearchCV(KNeighborsClassifier(), parameters, scoring = 'accuracy', cv = KFold(n_splits=5))
    knn.fit(X_train,y_train)

    # SVM
    parameters = {'C': range(1,11)}
    svc = GridSearchCV(svm.SVC(kernel = 'linear'), parameters, scoring = 'accuracy', cv = KFold(n_splits=5))
    svc.fit(X_train,y_train)

    # evaluate
    lr_test_acc = lr.score(X_test,y_test)
    lda_test_acc = lda.score(X_test,y_test)
    knn_test_acc = knn.best_estimator_.score(X_test,y_test)
    svc_test_acc= svc.best_estimator_.score(X_test,y_test)

    # print(lr_test_acc, lda_test_acc, knn_test_acc, svc_test_acc)
